### PR TITLE
add EH/SB tests to livetest pipeline

### PR DESCRIPTION
**Pull request by swathipil**
fixes: https://github.com/Azure/azure-sdk-for-python/issues/11588

### CODE REVIEW

File: .azure-pipelines/client.test.live.yml [hunk 1]
```diff
@@ -311,3 +310,590 @@ jobs:
...
+      - script: |
+          python --version
+          python -m pip install pytest
+          echo "Installing $DownloadArtifactWhl"
+          python -m pip install $(Pipeline.Workspace)/$(DownloadArtifactFolder)/$(DownloadArtifactWhl)
+          cd $(SdkPath)
+          python -m pip install --ignore-installed .
+          python -m pip install -r dev_requirements.txt
+          python -m pip list
+          pytest tests -v --doctest-modules --ignore=tests/perfstress_tests --junitxml=junit/eventhub/azure-eventhub/test-results-live.xml
```
Comment (scbedd) [inline .azure-pipelines/client.test.live.yml:418]: Don't abandon `python -m`! Everything is right up till here :D

**Review by scbedd — CHANGES_REQUESTED** [review]
All these jobs run pretty much exactly the same command. I think we you see where you have `strategy: ` set? And there is a single entry? I think we can put _all_ of these into the same single job definition that will be multiplied by the strategy alone. Can you point out anything that's honestly different between these runs that can't be covered by the variable? 

For your first attempt I'd just merge the matrix entries into a single one, and see what happens!  You can remove `MaxParallel` if you don't want them to run in serial :) 

Other than that single (tbh a bit of work) feedback, this looks good!

File: .azure-pipelines/client.test.live.yml [hunk 2]
```diff
@@ -311,3 +310,590 @@ jobs:
...
+      DevReqsPath: './azure-sdk-for-python/sdk/eventhub/azure-eventhub/dev_requirements.txt'
+      DownloadArtifactFolder: 'uamqp-win$(PythonArchitecture)-$(PythonVersion)-whl'
+      SdkPath: 'azure-sdk-for-python/sdk/eventhub/azure-eventhub'
+    steps:
+      - template: /.azure-pipelines/use-python-version.yml
+        parameters:
+          architecture: '$(PythonArchitecture)'
+          versionSpec: '$(PythonVersion)'
+      - script: |
+          git clone https://github.com/Azure/azure-sdk-for-python.git --single-branch
```
Comment (scbedd) [inline .azure-pipelines/client.test.live.yml:337]: We want something like a `sparse` checkout here.  I _think_ you can just add `-–depth 1` to your current command.

File: .azure-pipelines/client.test.live.yml [hunk 3]
```diff
@@ -311,3 +310,590 @@ jobs:
...
+        displayName: 'Clone azure-sdk-for-python'
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Downloading artifact'
+        inputs:
+          source: 'current'
+          patterns: '$(DownloadArtifactFolder)/*.whl'
+      - pwsh: |
+          $whlfile = Get-ChildItem -Filter *.whl $(Pipeline.Workspace)/$(DownloadArtifactFolder) | Select-Object -First 1 -ExpandProperty Name
+          echo "##vso[task.setvariable variable=DownloadArtifactWhl]$whlfile"
+          dir $(Pipeline.Workspace)/$(DownloadArtifactFolder)
```
Comment (scbedd) [inline .azure-pipelines/client.test.live.yml:347]: When you're running `pwsh`, `dir` is just aliased `Get-ChildItem`.

image

File: .azure-pipelines/client.test.live.yml [hunk 4]
```diff
@@ -311,3 +310,590 @@ jobs:
...
+        displayName: 'Get name of whl file to install'
+      - script: |
```
Comment (scbedd) [inline .azure-pipelines/client.test.live.yml:349]: NIT: is there a reason we're swapping between `pwsh` and `script` here?

Not that the commands really matter, but it's good to keep consistency.

File: .azure-pipelines/client.test.live.yml (hunk 1 above)
Comment (scbedd) [reply .azure-pipelines/client.test.live.yml:418]: (this same feedback is present for all the job definitions)

File: .azure-pipelines/client.test.live.yml (hunk 4 above)
Comment (swathipil) [reply .azure-pipelines/client.test.live.yml:349]: I just didn't want to think about the logic for script, so just copy-pasted the pwsh commands from a previous job :P
It's prob not too hard to translate so will fix 👍

File: .azure-pipelines/client.yml [hunk 5]
```diff
@@ -382,4 +382,4 @@ jobs:
         displayName: 'Publish wheel artifacts'
         inputs:
           artifactName: uamqp-$(Platform)-$(PythonVersion)-x64-whl
-          pathToPublish: 'wheelhouse'
+          pathToPublish: 'wheelhouse'
```
Comment (scbedd) [inline .azure-pipelines/client.yml:385]: nit: leave trailing newline! Good git etiquette

File: .azure-pipelines/test-eh-sb-sdk.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      AZURE_CLIENT_ID: $(aad-azure-sdk-test-client-id)
+      AZURE_CLIENT_SECRET: $(aad-azure-sdk-test-client-secret)
+      AZURE_TEST_RUN_LIVE: 'true'
+  - task: PublishTestResults@2
+    displayName: 'Publish test results'
+    condition: succeededOrFailed()
+    inputs:
+      testResultsFiles: '**/$(TestResultsPath)'
+      testResultsFormat: 'JUnit'
+      testRunTitle: '$(SDK) Python $(PythonVersion)'
```
Comment (scbedd) [inline .azure-pipelines/test-eh-sb-sdk.yml:59]: Nit: we probably want to include the OS Arch here right? (Windows/Linux/MacOs) given we are running this for all of them right?

**Review by scbedd — APPROVED** [review]
I like it a lot! Remember to uncomment the strategies before you merge!

File: .azure-pipelines/test-eh-sb-sdk.yml (hunk 6 above)
Comment (swathipil) [reply .azure-pipelines/test-eh-sb-sdk.yml:59]: ah yes, will do!

**Pull request merged**